virttest/test_setup.py

```diff
@@ -439,6 +439,12 @@ def __init__(self, params):
             target_hugepages = int(target_hugepages)
 
         self.target_hugepages = target_hugepages
+        self.target_nodes = params.get("target_nodes")
+        if self.target_nodes:
+            self.target_node_num = {}
+            for node in self.target_nodes.split():
+                num = params.object_params("node%s" % node).get("target_num")
+                self.target_node_num[node] = num
```

Comment:
Let us handle if user haven't provide target_num_node by distributing the target hugepages,
```python
if self.target_nodes:
    self.target_node_num = {}
    self.node_list = self.target_nodes.split()
    self.node_len = len(self.node_list)
    for node in node_list:
        num = params.object_params("node%s" % node).get("target_num")
        # let us distribute the hugepages if target_num_node is not set
        if not num:
            num = self.target_hugepages / self.node_len
            if node == node_list[0]:
                # if target_hugepages is not divisible by node_len
                # add remainder it to first node
                num += self.target_hugepages % self.node_len
        self.target_node_num[node] = num
```